### PR TITLE
style: track deviations from `prettier` defaults

### LINKED ISSUE #54402: style: formatting in new Next.js apps deviate from `prettier` defaults
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
$  npx --no-install next info

    Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP Fri Jan 27 02:56:13 UTC 2023
    Binaries:
      Node: 18.16.1
      npm: 9.7.2
      Yarn: 1.22.19
      pnpm: N/A
    Relevant Packages:
      next: 13.4.20-canary.2
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

_No response_

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/hamirmahal/nextjs-reproduce-style-inconsistency

### To Reproduce

1. Run `npx create-next-app@latest` from your terminal.
2. Go to `src/app/page.tsx`.
3. Save the file.
4. Notice that regardless of whether you set `prettier` to prefer single quotes or double quotes, there will be changes in the file.

https://github.com/vercel/next.js/assets/43425812/78f29485-421f-488a-b55a-0ae32281cd2f

### Describe the Bug

There's inconsistent styling in a brand new Next.js application. Either single quotation marks or double quotation marks should be preferred, as opposed to mixing both.

### Expected Behavior

Either single quotation marks or double quotation marks should be preferred.

### Which browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by hirehamir**
fixes #54402.

### CODE REVIEW

File: packages/create-next-app/templates/app-tw/ts/.prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "singleQuote": false
+}
```
Comment (hamirmahal) [inline packages/create-next-app/templates/app-tw/ts/.prettierrc.json:3]:
```suggestion
// Feel free to delete or ignore this file if you're not using `prettier`.
{
  "singleQuote": false
}
```
I wanted to add a comment like this but I think `.json` files don't allow comments.

Comment (hamirmahal) [inline packages/create-next-app/templates/app-tw/ts/.prettierrc.json:3]:
```suggestion
// Feel free to delete or ignore this file if you're not using `prettier`.
{
  "singleQuote": false
}
```
I wanted to add a comment like this, but `.json` files don't allow comments.

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 242881e2a4a036756411c5b14096d23b80b697f4

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by hamirmahal** [comment]
oh, gosh

**Review by styfle — CHANGES_REQUESTED** [review]
I'm not sure this is appropriate to add since the templates don't ship with `prettier` installed.

Also, if it was added to this template, it should be added to all templates.

**Comment by hamirmahal** [comment]
> Also, if it was added to this template, it should be added to all templates.

Ah, I agree. I addressed this just now.

**Comment by styfle** [comment]
> I wanted to add this change because I noticed fresh Next.js apps would have changes from formatting on save

Presumably if you added prettier as a dependency then you would also add the prettierrc file to configure as you wish.

**Comment by hamirmahal** [comment]
Hmm... do you think it makes more sense to ship `prettier` with Next.js?

**Comment by hamirmahal** [comment]
I was going for the smallest possible change that solves the original problem I was facing, which is that a fresh clone of Next.js will have changes just from doing Ctrl + S on a file, due to [Next.js templates not shipping with `prettier` defaults](https://github.com/vercel/next.js/blob/c8b2d1b44a9f9376eee7bc6c26e84bd7f5c03354/.prettierrc.json),

**Comment by styfle** [comment]
> Next.js will have changes just from doing Ctrl + S on a file,

Thats only if you have the prettier extension installed, right?

It would format differently if you used a different formatter on save.

**Comment by hamirmahal** [comment]
> Thats only if you have the prettier extension installed, right?

That's correct, yes.

**Comment by hamirmahal** [comment]
And it'll only happen if someone's using `prettier` with the default settings, unless we add the `.prettierrc` files from this pull request, in which case it won't happen at all.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Using overrides:
https://prettier.io/docs/en/configuration.html#configuration-overrides

I suggest we just override `packages/create-next-app/templates` to the defaults here: https://github.com/vercel/next.js/blob/canary/.prettierrc.json

**Comment by hamirmahal** [comment]
Are you saying to override `packages/create-next-app/templates/` so they conform to `prettier`'s default formatting?

**Comment by hamirmahal** [comment]
That'll be a much larger `diff` for this pull request, since we'd have to modify all the template files to match `prettier`'s default formatting, since the templates files all seem to currently conform to [`.prettierrc.json`](https://github.com/vercel/next.js/blob/canary/.prettierrc.json).

**Comment by hamirmahal** [comment]
Just bumping this again since I had to add

```
{
  "singleQuote": true,
  "semi": false
}
```

to a `.prettierrc.json` file in a new Next app to prevent any formatting changes when saving `app/page.tsx`

**Comment by hamirmahal** [comment]
> Using overrides: https://prettier.io/docs/en/configuration.html#configuration-overrides
> 
> I suggest we just override `packages/create-next-app/templates` to the defaults here: https://github.com/vercel/next.js/blob/canary/.prettierrc.json

I think that's what the files in this pull request are doing; please correct me if I'm mistaken.

**Comment by kachkaev** [comment]
👋 folks! WDYT of doing the opposite: formatting templates with default Prettier config? That would mean double quotes and semicolons. Quotes are a hot topic – https://github.com/prettier/prettier/issues/4102 (I am personally [in favour of double quotes](https://github.com/prettier/prettier/issues/4102#issuecomment-383016258)). The case for semicolons [is pretty straightforward](https://webreflection.medium.com/an-open-letter-to-javascript-leaders-regarding-no-semicolons-82cec422d67d): they simplify the life of newcomers and avoid confusion in a few edge cases. 

Next.js repo has its own history, so it’s fine to keep the existing preferences as is. But the templates is what a lot of folks interact with, so it’s best to keep the formatted using Prettier defaults.

**Comment by styfle** [comment]
> WDYT of doing the opposite, i.e. formatting templates with the default Prettier config

Thats probably a better idea because then we don't need the config.

However, are the default settings the same for prettier/dprint/biome/etc?

**Comment by hamirmahal** [comment]
> > WDYT of doing the opposite, i.e. formatting templates with the default Prettier config
> 
> Thats probably a better idea because then we don't need the config.

The pull request now uses the default `prettier` config.

**Comment by hamirmahal** [comment]
> > WDYT of doing the opposite, i.e. formatting templates with the default Prettier config
> 
> Thats probably a better idea because then we don't need the config.

This pull request now uses the default `prettier` config.

**Comment by JounQin** [comment]
> Thats probably a better idea because then we don't need the config.

> However, are the default settings the same for prettier/dprint/biome/etc?

That's a good point, and prettier was born earlier, so I believe other formatting tools should adapt prettier's defaults, instead of reversing.

**Pull request closed without merging**